tests/unit/components/SharedComponents/InlineUser/InlineUser.test.js

```diff
@@ -22,16 +22,32 @@ jest.mock( "@react-navigation/native", ( ) => {
 const mockUser = factory( "RemoteUser" );
 const mockUserWithoutImage = factory( "RemoteUser", { icon_url: null } );
 
+const consistentUser = { login: "some_login", icon_url: "some_icon_url" };
+
+jest.mock(
+  "components/SharedComponents/UserIcon/UserIcon",
+  () => function MockUserIcon( ) {
+    const MockName = "mockUserIcon";
+    return <MockName testID={MockName} />;
+  }
+);
+
 describe( "InlineUser", ( ) => {
+  it( "renders reliably", () => {
+    // Snapshot test
+    render( <InlineUser user={consistentUser} /> );
+    // TODO: Enable this test when Typography is used instead of Text
+    // expect( screen ).toMatchSnapshot();
+  } );
+
   it( "displays user handle and image correctly", async ( ) => {
     render( <InlineUser user={mockUser} /> );
     // Check for user name text
     expect( screen.getByText( `@${mockUser.login}` ) ).toBeTruthy( );
     // This image appears after useIsConnected returns true
     // so we have to use await and findByTestId
-    const profilePicture = await screen.findByTestId( "UserIcon.photo" );
+    const profilePicture = await screen.findByTestId( "mockUserIcon" );
```

Comment:
What's the reason for querying by testID throughout this component? It's still my understanding that the best way to test for accessibility is by looking for elements the way the user will see and interact with them on screen.

Reply:
Yes, it was my understanding that when I want to have a container take all the accessibility props, I should not give the children any accessibility props (but I can look into that again). So, there was nothing else really available to querying for for the images, following https://testing-library.com/docs/queries/about/#priority until the end to byTestId